src/main/java/rx/exceptions/Exceptions.java

```diff
@@ -76,12 +76,7 @@ public static void throwIfFatal(Throwable t) {
         if (t instanceof OnErrorNotImplementedException) {
             throw (OnErrorNotImplementedException) t;
         } else if (t instanceof OnErrorFailedException) {
-            Throwable cause = t.getCause();
-            if (cause instanceof RuntimeException) {
-                throw (RuntimeException) cause;
```

Comment:
I don't think that there is such code. Such unwrapped exceptions were swallowed most of the time anyway.